tests/v2/actions/etcdsnapshot/verify.go

```diff
@@ -16,25 +15,59 @@ import (
 	kwait "k8s.io/apimachinery/pkg/util/wait"
 )
 
-// VerifySnapshots verifies that all snapshots come to an active state, and the correct number
+const (
+	expectedRKE1SnapshotLen = 1
+)
+
+// VerifyRKE1Snapshots verifies that all snapshots come to an active state, and the correct number
+// of snapshots were taken based on the number of nodes and snapshot type (s3 vs. local)
+func VerifyRKE1Snapshots(client *rancher.Client, clusterName string, snapshotIDs []string) error {
+	err := kwait.PollUntilContextTimeout(context.TODO(), 5*time.Second, extdefault.FiveMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
+		var snapshotObj any
+		var state string
+
+		for _, snapshotID := range snapshotIDs {
+			snapshotObj, err = client.Management.EtcdBackup.ByID(snapshotID)
+			state = snapshotObj.(*management.EtcdBackup).State
+
+			if err != nil {
+				return false, nil
+			}
+
+			if state != "active" {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// RKE1 doesn't give us snapshot-per-node. It gives us one object representing the snapshot for the cluster.
+	if expectedRKE1SnapshotLen != len(snapshotIDs) {
+		logrus.Info(snapshotIDs)
+		return fmt.Errorf("unexpected number of snapshots. Expected %v but have %v", expectedRKE1SnapshotLen, len(snapshotIDs))
+	}
+
+	return nil
+}
+
+// VerifyV2ProvSnapshots verifies that all snapshots come to an active state, and the correct number
 // of snapshots were taken based on the number of nodes and snapshot type (s3 vs. local)
-func VerifySnapshots(client *rancher.Client, clusterName string, snapshotIDs []string, isRKE1 bool) error {
+func VerifyV2ProvSnapshots(client *rancher.Client, clusterName string, snapshotIDs []string) error {
 	isS3 := false
 	err := kwait.PollUntilContextTimeout(context.TODO(), 5*time.Second, extdefault.FiveMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
 		var snapshotObj any
 		var state string
 
 		for _, snapshotID := range snapshotIDs {
-			if isRKE1 {
-				snapshotObj, err = client.Management.EtcdBackup.ByID(snapshotID)
-				state = snapshotObj.(*management.EtcdBackup).State
-			} else {
-				snapshotObj, err = client.Steve.SteveType(shepherdsnapshot.SnapshotSteveResourceType).ByID(snapshotID)
-				state = snapshotObj.(*steveV1.SteveAPIObject).ObjectMeta.State.Name
-
-				if strings.HasSuffix(snapshotObj.(*steveV1.SteveAPIObject).Name, "-s3") {
-					isS3 = true
-				}
+			snapshotObj, err = client.Steve.SteveType(shepherdsnapshot.SnapshotSteveResourceType).ByID(snapshotID)
+			state = snapshotObj.(*steveV1.SteveAPIObject).ObjectMeta.State.Name
+
+			store, ok := snapshotObj.(*steveV1.SteveAPIObject).Annotations["etcdsnapshot.rke.io/storage"]
+			if ok && store == "s3" {
```

Comment:
why do we want to include this check in the poll iteration? 
 Also, wouldn’t isS3 be reset to false if, for example, we have 5 snapshots and the last one is local, even if an earlier snapshot is stored in S3? 

I’m trying to understand why would we want to check this for each snapshot ID?

Reply:
1. because we need to set if this is s3 or not, and the only way to do that is either pass it in as a var or do a simple if statement like this, or possibly do a separate for loop outside the poll but that seemed redundant since we already are getting the full list here.
2. no, this check only sets to true. False is only set on init of the var
3. I could add a break if need be so that as soon as it hits, it doesn't check the rest.

Reply:
got it. yeah I'm cool with having the break added so that it doesnt have to verify for every object

Reply:
spoke offline adding a break would cause the status check to be disregarded for the rest. so we can proceed with current functionality.